ci: add front code formatting check

emersion opened the pull request:
We run ESLint already at build-time, however we never run Prettier. As a result, non-TS files such as SCSS are not getting checked.

* * *

Of course, this isn't enough as-is because Prettier will trip on many SCSS files.

emersion committed:
ci: add front code formatting check

We run ESLint already at build-time, however we never run Prettier.
As a result, non-TS files such as SCSS are not getting checked.

Yohh reviewed (approved):
nice! thanks :muscle:

emersion commented:
List of re-formatted files: https://github.com/OpenRailAssociation/osrd/actions/runs/9854031436/job/27205971674?pr=7988

I've pushed a commit to re-format everything but it needs more eyeballing. For instance, `src/reducers/osrdconf/infra_schema.json` is automatically generated (we don't want to fail CI when re-generating the file).

emersion committed:
front: add .prettierignore

Ignore formatting for all automatically-generated and
machine-generated files.

emersion commented:
I've added a `.prettierignore` file with machine-generated files.

emersion commented:
Ah, maybe we want to ignore map styles as well. These seem to be machine-generated. Edited the commits to do so.

Khoyo commented on front/.prettierignore:
```diff
@@ -0,0 +1,7 @@
+src/modules/trainschedule/components/ImportTrainSchedule/rollingstock_opendata2osrd.json
+src/reducers/osrdconf/infra_schema.json
+src/assets/mapstyles/*.json
+src/utils/__tests__/assets/*.json
```
I'd add `src/config/openapi-editoast-config.ts`  and `src/config/openapi-gateway-config.ts` too as they are generated

emersion replied:
Hm, I don't think these are generated? These are hand-written config files to configure the OpenAPI codegen, I believe?

anisometropie replied:
yeah `src/config/openapi-editoast-config.ts` and `src/config/openapi-gateway-config.ts` are not generated, 
the ones to ignore are `src/common/api/generatedEditoastApi.ts` and `src/common/api/osrdGatewayApi.ts`.

They are also ignored by eslint (even though some eslint is run **once** when they are generated)

emersion replied:
Right. Currently `scripts/generate-types.sh` runs `eslint` after generating the files, so it's not strictly necessary to add these files to the ignore list.

anisometropie replied:
we don’t really care about the formatting of these files. eslint is run just once at generation.
otherwise they are ignored in the eslint patterns, I would ignore these too in prettier.

Khoyo replied:
Right, I copy pasted the wrong thing ^^'

emersion replied:
@anisometropie That makes sense, I added these files for consistency.

Khoyo reviewed (approved):
Otherwise LGTM

flomonster commented on front/README.md:
```diff
@@ -40,26 +40,27 @@ Launches end to end tests.
-> [!CAUTION]
-> If you try to run `yarn start` instead of running it through docker, you'll notice it doesn't
-> work because the gateway can't access your local port from inside a container. 2 solutions:
+> [!CAUTION] If you try to run `yarn start` instead of running it through docker, you'll notice it
+> doesn't work because the gateway can't access your local port from inside a container. 2
+> solutions:
```
This is broken (you can check the GitHub preview). The line feed is required.
We should probably add this file to the `.prettierignore`.

emersion replied:
Eh :/ 

Oh well, I just added a rule to ignore `*.md` for now.

(An alternative would be to add two spaces at the end of the line probably, but we'd risk getting it removed by mistake by some editors stripping whitespace at end-of-lines automatically when saving a file.)

flomonster reviewed (approved):
LGTM (not tested)

emersion committed:
front: tighten up ignoredPatterns in .eslintrc

The pattern "src/common/api/*Api.ts" was a bit too coarse: some
hand-written files were being ignored as well:

    > ls src/common/api/*Api.ts
    src/common/api/documentApi.ts           src/common/api/osrdEditoastApi.ts
    src/common/api/generatedEditoastApi.ts  src/common/api/osrdGatewayApi.ts
    src/common/api/graouApi.ts

We only want to ignore machine-generated files here, so list these
two files explicitly instead.

emersion commented:
While resolving the comments from @anisometropie, I noticed that a few hand-written files were being ignored in the ESLint config. I've fixed this in a new commit: "front: tighten up ignoredPatterns in .eslintrc". See its commit message for details.

emersion committed:
front: re-format all files

Run `yarn prettier . --write`.

emersion commented:
Does this version still look good with the additional commit?

anisometropie reviewed (approved):
Great, thanks for tidying all that up.

clarani reviewed (approved):
LGTM
